pkg/datagatherer/k8s/fieldfilter_test.go

```diff
@@ -18,10 +18,10 @@ func TestSelect(t *testing.T) {
 			"metadata": map[string]interface{}{
 				"name":      "example",
 				"namespace": "example",
-				"annotations": map[string]string{
+				"annotations": map[string]interface{}{
```

Comment:
None of the maps should be of type `map[string]string{}`, kubernetes uses yaml.v2 which always produces `map[string]interface{}{}` maps.

yaml.v3 would produce `map[string]string{}`, but Kubernetes does not use that version and we expect the unstructured helpers to get updated once they switch.